### ISSUE TITLE
Langversion default not respected with net10 sdk installed

### ISSUE THREAD

**Issue by cthompson92** [dotnet/roslyn#81189]
### Describe the bug
After installing VS2026 and the net10 SDK, I am seeing unexpected build errors when compiling both in Visual Studio (2022 and 2026) and with the dotnet command line, for a project set up to target net8.0 and using `LangVersion default`. The build errors seem to be directly related to the changes in C#14.

### To Reproduce

Clone [my language-ext fork](https://github.com/cthompson92/language-ext-ef-svc-sample/commit/f369e94fcaccfa0d79fea684bf641c4326a96730)

`dotnet build -c Release`

### Exceptions (if any)

Unexpected compiler errors:
> \LanguageExt.Core\Immutable Collections\Arr\Arr.Module.cs(110,24): error CS0023: Operator '.' cannot be applied to operand of type 'void'
> \LanguageExt.Core\Immutable Collections\Arr\Arr.cs(605,29): error CS0023: Operator '.' cannot be applied to operand of type 'void'

These errors seem to be directly tied to compiling with C# 14, as the project compiles fine if I explicitly request C#12 or C#13:
`dotnet build -c Release -p:LangVersion=13.0`
`dotnet build -c Release -p:LangVersion=12.0`

While `dotnet build -c Release -p:LangVersion=14.0` results in the same set of errors.

My understanding is that when the projects are targeting `net8.0` and specifying `LangVersion=default` then C#12.0 should be used to compile, regardless of any higher C# versions/dotnet SDKs installed locally.

### Further technical details
<details><summary>details of dotnet --info</summary>
<p>
dotnet --info
.NET SDK:
 Version:           10.0.100
 Commit:            b0f34d51fc
 Workload version:  10.0.100-manifests.355811b7
 MSBuild version:   18.0.2+b0f34d51f

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.26100
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\10.0.100\

.NET workloads installed:
There are no installed workloads to display.
Configured to use workload sets when installing new manifests.
No workload sets are installed. Run "dotnet workload restore" to install a workload set.

Host:
  Version:      10.0.0
  Architecture: x64
  Commit:       b0f34d51fc

.NET SDKs installed:
  9.0.306 [C:\Program Files\dotnet\sdk]
  10.0.100 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 8.0.21 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 9.0.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 10.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 8.0.21 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 9.0.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 10.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 8.0.21 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 9.0.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 10.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
</p>
</details>
- The IDE (VS / VS Code/ VS4Mac) you're running on, and its version

Tested in both VS2026 and VS2022, as well as at the command line

I also tested by adding this global.json to the repo and running `dotnet build -c Release` again, and it compiles successfully. I think this further indicates that this issue is specific to the net10 SDK.
```json
{
    "sdk": {
        "version": "9.0.0",
        "rollForward": "latestMinor"
    }
}
```

**Comment by jjonescz** [comment]
> My understanding is that when the projects are targeting `net8.0` and specifying `LangVersion=default` then C#12.0 should be used to compile, regardless of any higher C# versions/dotnet SDKs installed locally.

That's incorrect. LangVersion=default means "the latest version the compiler supports" ([docs](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version)). If you are using .NET 10 SDK, the compiler supports C# 14 as the latest lang version. If you want to stay at C# 12.0, you can set `LangVersion=12.0` explicitly.

> error CS0023: Operator '.' cannot be applied to operand of type 'void'

That's most likely this change: https://github.com/dotnet/roslyn/blob/main/docs/compilers/CSharp/Compiler%20Breaking%20Changes%20-%20DotNet%2010.md#enumerablereverse
However, using LangVersion=14.0 with net8.0 is an unsupported configuration. Targeting net10.0 should work fine.
